src/components/structured-text-block/structured-text-block.vue

```diff
@@ -64,6 +105,7 @@
           class: 'structured-text__highlighted-list-item',
         }, h(StructuredTextBlock, {
           content: listItem.body,
+          onUpdateTocItems: updateTocItems,
```

Comment:
I would prefer not to use side effects on the rendering methods to construct the TOC items and use a computed property for determining the toc items.

I think this could take some time, so for speed I'll create a separate ticket to see if we can refactor as I think Vera would really like to have this feature available before her holiday.